src/utils/orcid.py

```diff
@@ -113,3 +113,34 @@ def get_orcid_record_details(orcid):
             details["country"] = affiliation["address"]["country"]
 
     return details
+
+
+def encode_state(next_url, orcid_action):
+    """
+    Encode information related to Janeway application state in a form that can
+    be sent and received back from ORCiD authentication, independently
+    of the redirect URI, which the ORCiD auth system must be able to predict.
+    See https://info.orcid.org/documentation/integration-guide/customizing-the-sign-in-register-screen/#h-identify-the-researcher-by-a-custom-state-parameter
+
+    :param next_url: a raw URL (can include its own query parameters)
+    :param orcid_action: 'login' or 'register'
+    :return: a query string with percent-encoded paramater values
+        that can be passed to ORCiD in a URL as a callback
+    """
+    querydict = QueryDict(mutable=True)
+    if next_url:
+        querydict['next'] = next_url
+    if orcid_action:
```

Comment:
This code is fairly defensive for what seem to be non-nullable arguments, why is this?

Reply:
I was trying to avoid empty values, since that would result in `next=&action=`, which would then populate query dicts like `{'next': None, 'action': None}` which are a bit more bug prone to use in the views.